### ISSUE TITLE
Upload script returns non-success exit code when successful

### ISSUE THREAD

**Issue by coop** [nerves-project/nerves_firmware_ssh#50]
### Environment

* Elixir version (`elixir -v`): `Elixir 1.9.0 (compiled with Erlang/OTP 22)`
* Nerves environment: (`mix nerves.env --info`):

```
$ mix nerves.env --info
|nerves_bootstrap| Environment Package List

  No packages found
|nerves_bootstrap| Loadpaths Start


Nerves environment
  MIX_TARGET:   host
  MIX_ENV:      dev

NERVES_SYSTEM is unset
NERVES_TOOLCHAIN is unset
|nerves_bootstrap| Environment Variable List
  target:     host
  toolchain:  unset
  system:     unset
  app:        /Users/coop/Code/asdf/fw

|nerves_bootstrap| Loadpaths End
```

* Additional information about your host, target hardware or environment that
  may help

### Current behavior

`upload.sh` returns a non-success exit code when successful:

```
$ MIX_TARGET=rpi3 MIX_ENV=prod ./upload.sh nerves.local; echo $?
Path: ./_build/rpi3_prod/nerves/images/asdf.fw
Product: asdf 0.1.0
UUID: 3855b619-5f23-5ef4-7775-75592cae471d
Platform: rpi3

Uploading to nerves.local...
Running fwup...
fwup: Upgrading partition A
|====================================| 100% (24.51 / 24.51) MB
Success!
Elapsed time: 4.809 s
Rebooting...
Received disconnect from 192.168.1.130 port 8989:11: Terminated (shutdown) by supervisor
Disconnected from 192.168.1.130 port 8989
255
```

The upload succeeds and the device correctly reboots but the exit code is 255. Mostly this isn't a problem except I'm pushing updates with Ansible and it expects a success exit code - I can always wrap upload in another script that catches the "bad" exit code and returns an OK.

### Expected behavior

According to the documentation the exit code should reflect the successfulness of the command:

```
# 5. The ssh connection is closed with an exit code to indicate success or
#    failure
```

**Comment by fhunleth** [comment]
Thanks for reporting this. I've started seeing that the device reboot beats the close on the ssh connection too.

**Comment by coop** [comment]
@fhunleth I ended up grepping for Success in the output:

```
printf "fwup:$FILESIZE,reboot\n" | cat - $FILENAME | ssh -s -p 8989 $SSH_OPTIONS $DESTINATION nerves_firmware_ssh | grep -1 'Success!'
```

**Comment by fhunleth** [comment]
I think that I may have fixed this one in v0.4.5. The ssh connection wasn't getting properly terminated after the reboot and this started causing other issues (maybe a new version of Erlang brought out this new behavour? Don't know why I hadn't seen it before).
